source/cellular_at_core.c

```diff
@@ -766,11 +765,21 @@ CellularATError_t Cellular_ATStrDup( char ** ppDst,
     char * p = NULL;
     CellularATError_t atStatus = CELLULAR_AT_SUCCESS;
     const char * pTempSrc = pSrc;
+    CellularATStringValidationResult_t stringValidationResult = CELLULAR_AT_STRING_UNKNOWN;
 
-    if( ( ppDst == NULL ) || ( pTempSrc == NULL ) || ( strnlen( pTempSrc, CELLULAR_AT_MAX_STRING_SIZE ) >= CELLULAR_AT_MAX_STRING_SIZE ) )
+    if( ( ppDst == NULL ) || ( pTempSrc == NULL ) )
     {
         atStatus = CELLULAR_AT_BAD_PARAMETER;
     }
+    else
+    {
+        validateString( pTempSrc, &stringValidationResult );
+
+        if( stringValidationResult != CELLULAR_AT_STRING_VALID )
+        {
+            atStatus = CELLULAR_AT_BAD_PARAMETER;
+        }
+    }
```

Comment:
Just to keep it similar to the other functions in the file:

```suggestion
    if( atStatus == CELLULAR_AT_SUCCESS )
    {
        validateString( pTempSrc, &stringValidationResult );

        if( stringValidationResult != CELLULAR_AT_STRING_VALID )
        {
            atStatus = CELLULAR_AT_BAD_PARAMETER;
        }
    }
```

Reply:
Update in new commit.